pkgs/development/tools/documentation/doxygen/default.nix

```diff
@@ -1,15 +1,14 @@
-{ stdenv, cmake, fetchurl, python3, flex, bison, qt4, CoreServices, libiconv }:
+{ stdenv, cmake, fetchFromGitHub, python3, flex, bison, qt5, CoreServices, libiconv }:
 
 stdenv.mkDerivation rec {
-
-  name = "doxygen-1.8.19";
-
-  src = fetchurl {
-    urls = [
-      "mirror://sourceforge/doxygen/${name}.src.tar.gz" # faster, with https, etc.
-      "http://doxygen.nl/files/${name}.src.tar.gz"
-    ];
-    sha256 = "1lvqfw2yzba588c5ggl8yhw7aw4xkk44mrghsd9yqlajc48x25dc";
+  pname = "doxygen";
+  version = "1.8.20";
+
+  src = fetchFromGitHub {
+    owner = "doxygen";
+    repo = "doxygen";
+    rev = "Release_${stdenv.lib.replaceStrings [ "." ] [ "_" ] version}";
+    sha256 = "17chvi3i80rj4750smpizf562xjzd2xcv5rfyh997pyvc1zbq5rh";
```

Comment:
GitHub is preferred to upstream tarballs?

Reply:
Not sure if we have any policy on that kind of thing, but personally, I prefer downloading tags from a git repository vs. tarballs from random project sites. Plus, that way update bots like @r-ryantm can potentially pick up new releases.

Reply:
The bot doesn't care either way for updates. You can see there are many bot updates of this package in the history: https://github.com/NixOS/nixpkgs/commits/1f75b60c56688d79366ceb7cd11b597fc680dbfd/pkgs/development/tools/documentation/doxygen/default.nix

Reply:
Right, nevermind that part then. I can also revert to fetchurl, if that's preferred, although I can see arguments for and against both ways.

Reply:
Personally, I trust GitHub more than sourceforge and the canonical file mirror isn't https, so I like this change.

Reply:
OK.

I managed to get https from the canonical URL, though maybe there's something weird about it as my `nix-prefetch-url` wouldn't stomach it.